src/local-fonts/upload-font-form.js

```diff
@@ -0,0 +1,239 @@
+
+				setFormData( {
+					file,
+					name: fontName,
+					weight: fontWeight,
+					style: isItalic ? 'italic' : 'normal',
+					variable: isVariable,
+					axes,
+				} );
+			};
+		};
+	};
+
+	const resetDefaultVariableSettings = () => {
+		const newAxes = Object.keys( formData.axes ).reduce(
+			( acc, axisTag ) => {
+				acc[ axisTag ] = {
+					...formData.axes[ axisTag ],
+					currentValue: formData.axes[ axisTag ].defaultValue,
+				};
+				return acc;
+			},
+			{}
+		);
+		setFormData( {
+			...formData,
+			axes: newAxes,
+		} );
+	};
+
+	const fontVariationSettings = variableAxesToCss( formData.axes );
+
+	return (
+		<>
+			<form
+				method="POST"
+				id="font-upload-form"
+				action=""
+				encType="multipart/form-data"
+			>
+				<input type="hidden" name="nonce" value={ nonce } />
+
+				<div className="form-group">
+					<label htmlFor="font-file">
+						{ __( 'Font file:', 'create-block-theme' ) }
+					</label>
+					<input
+						type="file"
+						name="font-file"
+						id="font-file"
+						onChange={ onFileSelectChange }
+						accept=".otf, .ttf, .woff, .woff2"
+					/>
+					<small>
+						{ __(
+							'.otf, .ttf, .woff, .woff2 file extensions supported',
+							'create-block-theme'
+						) }
+					</small>
+				</div>
+
+				<h4>
+					{ __(
+						'Font face definition for this font file:',
+						'create-block-theme'
+					) }
+				</h4>
+
+				<div className="form-group">
+					<InputControl
+						label={ __( 'Font name:', 'create-block-theme' ) }
+						type="text"
+						name="font-name"
+						id="font-name"
+						placeholder={ __( 'Font name', 'create-block-theme' ) }
+						required
+						value={ formData.name }
+						onChange={ ( val ) =>
+							setFormData( { ...formData, name: val } )
+						}
+					/>
+				</div>
+
+				<div className="form-group">
+					<SelectControl
+						label={ __( 'Font style:', 'create-block-theme' ) }
+						name="font-style"
+						id="font-style"
+						required
+						value={ formData.style }
+						onChange={ ( val ) =>
+							setFormData( { ...formData, style: val } )
+						}
+					>
+						<option value="normal">Normal</option>
+						<option value="italic">Italic</option>
+					</SelectControl>
+				</div>
+
+				<div className="form-group">
+					<InputControl
+						label={ __( 'Font weight:', 'create-block-theme' ) }
+						type="text"
+						name="font-weight"
+						id="font-weight"
+						placeholder={ __(
+							'Font weight:',
+							'create-block-theme'
+						) }
+						required
+						value={ formData.weight }
+						onChange={ ( val ) =>
+							setFormData( { ...formData, weight: val } )
+						}
+					/>
+				</div>
+
+				{ formData.variable && (
+					<input
+						type="hidden"
+						name="font-variation-settings"
+						value={ fontVariationSettings }
+					/>
+				) }
+			</form>
+
+			{ /* Render the range controls for each available axis of a variable font */ }
+			{ formData.variable && (
+				<div className="variable-settings">
```

Comment:
I think it might make more sense if the axis controls for the variable font were incorporated into the preview settings somehow. Where they are placed right now feels like that by changing the variable font axis you are somehow changing the settings that font will be uploaded to the theme with. 

Suggestion:
<img width="1624" alt="Screenshot 2023-03-14 at 10 43 31 AM" src="https://user-images.githubusercontent.com/5375500/224961579-eb32cc8d-7182-4a33-b466-d0dbf7ab2f96.png">

You could also just have one "reset" button that way too.

Reply:
Good point! I moved the variable fonts preview options to the demo option preview panel